python/plugins/processing/algs/qgis/ExecuteSQL.py

```diff
@@ -125,7 +129,21 @@ def processAlgorithm(self, parameters, context, feedback):
 
         df = QgsVirtualLayerDefinition()
         for layerIdx, layer in enumerate(layers):
-            df.addSource('input{}'.format(layerIdx + 1), layer.id())
+
+            # Issue https://github.com/qgis/QGIS/issues/24041
+            # When using this algorithm from the graphic modeler, it may try to
+            # access (thanks the QgsVirtualLayerProvider) to memory layer that
+            # belongs to temporary QgsMapLayerStore, not project.
+            # So, we write them to disk is this is the case.
+            if not QgsProject.instance().mapLayer(layer.id()):
+                suffix = "." + QgsVectorFileWriter.supportedFormatExtensions()[0]
```

Comment:
Why not hardcode the geopackage extension here?

Reply:
See my comment above and the other PR https://github.com/qgis/QGIS/pull/30442#discussion_r298502704 

GPKG is optional for GDAL.